src/Containers/FileUpload/FileUpload.tsx

```diff
@@ -0,0 +1,456 @@
+import React,{useCallback,useRef,useEffect,useLayoutEffect,useReducer} from 'react'
+import styled from 'styled-components'
+import {flex} from '@Utils/Mixins'
+import {TextLayout} from '@Layouts'
+import {useDropzone} from 'react-dropzone'
+import {Image} from '@styled-icons/fa-solid/Image';
+import {CheckCircle} from '@styled-icons/fa-solid/CheckCircle'
+import {TimesCircle} from '@styled-icons/fa-solid/TimesCircle'
+import {MainTheme} from '@Themes'
+import {Loading} from '../Loading/Loading'
+import {BottomPanel} from './BottomPanel'
+import {Container,Icon} from './StyledIcons'
+import {FileMovingAnimation} from './FileMovingAnimation'
+import {IsFailureIsSuccessPanel} from './IsFailureIsSuccessPanel'
+
+interface IOptions{
+    position:boolean;
+    opacity:number;
+}
+
+interface IState{
+    height:number|undefined;
+    totalHeight:number;
+    totalHeightPlus:number;
+    padding:number;
+    margin:number;
+    maxHeight:number|undefined;
+    loading:IOptions;
+    isSuccess:IOptions;
+    isFailure:IOptions;
+    positionTopLoading:number;
+    loadingContainerHeight:number;
+    componentWidth:number;
+    isSuccessWidth:number;
+    isDragEnter:boolean;
+}
+
+const SET_HEIGHT='SET_HEIGHT'
```

Comment:
What is this?

Reply:
this is the redux way to declare types of actions